src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/AdditionalFactorsForWomen.kt

```diff
@@ -66,25 +65,19 @@ class AdditionalFactorsForWomen(
     }
 
   private fun getArsonOrViolencePoints(convictions: Collection<Conviction>): Int =
-    if (convictions.flatMap { it.offences }
-      .map { it.offenceDetail }.any {
-        OFFENCE_CODES.contains(it.mainCategoryCode)
-      }
-    ) 2 else 0
-
-  private fun getSentenceLengthPoints(convictions: Collection<Conviction>): Int {
+    if (convictions.flatMap { it.offenceMainCategoryCodes }.any { OFFENCE_CODES.contains(it) }) 2 else 0
 
-    val custodialSentences = convictions.map { it.sentence }.filter { isCustodial(it) }
-    val custodialConvictions = convictions.filter { it.sentence in custodialSentences }
+  private fun getSentenceLengthPoints(convictions: Collection<Conviction>): Int =
+    if (convictions
+      .filter { it.sentence.isCustodial() }
+      .any { it.sentence.isTenMonthsOrOver() || it.isIndeterminate() }
+    ) 2 else 0
```

Comment:
😎

Reply:
Domain objects showing a big readability benefit here

Reply:
its what Andy Marke said, the top of file is really nice at the expense of the bottom